src/main/kotlin/onm/configuration/ConfigurationDataClass.kt

```diff
@@ -0,0 +1,52 @@
+package onm.configuration
+
+/**
+ * Data class used as configuration part for out framework.
+ * */
+data class ConfigurationDataClass(
+        /**
+         * structure of this map is like that:
+         *
+         * ```RoomType: DeviceType - count```
+         *
+         * so example:
+         *
+         * ```Kitchen : Fridge : 1```
+         * */
+        val roomsAndDevices: Map<RoomConfig, Collection<DeviceConfig>> //todo add more config options
+)
+
+/**
+ * Represents room config.
+ * */
+data class RoomConfig(
+        val roomType: RoomType,
+        val description: String,
+        val floor: Int
+)
+
+/**
+ * Represents device config.
+ * */
+data class DeviceConfig(
+        val deviceType: DeviceType,
+        val powerConsumption: PowerConsumption
+)
+
+/**
+ * Data class which represents power consumptions in different devices states.
+ * */
+data class PowerConsumption(
+        val idleState: Int? = null,
+        val workingState: Int? = null,
+        val turnedOffState: Int? = null
+) {
+    companion object {
+        /**
+         * Creates power consumption with default parameters
+         * */
+        fun defaultPowerConsumption(): PowerConsumption {
+            return PowerConsumption()
+        }
```

Comment:
nope, this just use constructor and return empty data class
It is better to use it as static factory method because of better code readability -> you know that you will be using default data

we can adjust it that this method will take enum `DeviceType` as parameter and use default values.... that would be probably better, i will refactor that

Reply:
addition -> it is not possible to do that way

what if user want to specify only idle power consumption? -> we need to check nullable type when we are creating instance `DeviceState`